### ISSUE TITLE
Is the categorical one implemented right?

### ISSUE THREAD

**Issue by ysyyork** [umbertogriffo/focal-loss-keras#13]
I think the cross entropy part is wrong. 
```
def categorical_focal_loss_fixed(y_true, y_pred):
        """
        :param y_true: A tensor of the same shape as `y_pred`
        :param y_pred: A tensor resulting from a softmax
        :return: Output tensor.
        """

        # Scale predictions so that the class probas of each sample sum to 1
        y_pred /= K.sum(y_pred, axis=-1, keepdims=True)

        # Clip the prediction value to prevent NaN's and Inf's
        epsilon = K.epsilon()
        y_pred = K.clip(y_pred, epsilon, 1. - epsilon)

        # Calculate Cross Entropy
        cross_entropy = -y_true * K.log(y_pred)

        # Calculate Focal Loss
        loss = alpha * K.pow(1 - y_pred, gamma) * cross_entropy

        # Compute mean loss in mini_batch
        return K.mean(K.sum(loss, axis=-1))

    return categorical_focal_loss_fixed
```
So currently  `cross_entropy = -y_true * K.log(y_pred)`. This already ignored all the true negative cases. I think the right implementation should be below:
```
...
loss = - y_true * K.log(y_pred) * K.pow(1 - y_pred, gamma) * alpha \
                        - (1 - y_true) * K.log(1 - y_pred) * K.pow(y_pred, gamma) * (1 - alpha)
...
```
The loss should be implemented like this. Let me know if this makes sense. When I'm testing using this loss on my model. this new one works much better than the old one.

**Comment by ysyyork** [comment]
NVM this assumes softmax. so the original implementation should be ok
